### DISCUSSION TITLE
Affine behind Traefik

### DISCUSSION THREAD

**Discussion by stonecrumbler** [toeverything/AFFiNE#7248] [General]
Hi there! I'm running a home server with a couple of services in docker containers, exposing them via Traefik on separate sub-domains. 

With Affine, however,  I'm unable to make it work, no matter what I try. I added the Traefik labels to the compose file for Affine, and set up everything as I have with my other services. 

It seems like Affine doesn't respond when the request is proxied, but works fine if i access it using the default port. 

Is there a trick to make it work with a proxy?

**Comment by gnacoding** [comment]
Hi

I had to add these environments to the docker-compose:
      - AFFINE_SERVER_HOST=sub.domain.com
      - AFFINE_SERVER_PORT=80 //this was the culprit for me as affine deploys on port 3010 i think, so i had to override it
      - AFFINE_SERVER_HTTPS=true //optional, only if you have a SSL Cert


traefik labels:

      - "traefik.enable=true"
      - "traefik.http.routers.affine-https.rule=Host(`sub.domain.com`)"
      - "traefik.http.routers.affine-https.entrypoints=websecure-https" //entrypoint might differ from yours, check your traefik config
      - "traefik.http.routers.affine-https.tls=true"
      - "traefik.http.routers.affine-https.service=affine-https"
      - "traefik.http.services.affine-https.loadbalancer.server.port=80"
      - "traefik.http.routers.affine-https.tls.certresolver=certresolver"

**Reply by stonecrumbler** [reply]
Thanks for the help. I managed to get it working. The main problem was that I somehow forgot to add the network to the affine container, so traefik could not connect to it. Such a stupid mistake on my part. 🤯
